sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2025 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package io.dapr.it.testcontainers.workflows.crossapp;
+
+import io.dapr.testcontainers.Component;
+import io.dapr.testcontainers.DaprContainer;
+import io.dapr.testcontainers.DaprLogLevel;
+import io.dapr.testcontainers.DaprPlacementContainer;
+import io.dapr.testcontainers.DaprSchedulerContainer;
+import io.dapr.workflows.client.DaprWorkflowClient;
+import io.dapr.workflows.client.WorkflowInstanceStatus;
+import io.dapr.workflows.client.WorkflowRuntimeStatus;
+import io.dapr.config.Properties;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.MountableFile;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.io.File;
+
+import static io.dapr.it.testcontainers.ContainerConstants.DAPR_RUNTIME_IMAGE_TAG;
+import static io.dapr.testcontainers.DaprContainerConstants.DAPR_PLACEMENT_IMAGE_TAG;
+import static io.dapr.testcontainers.DaprContainerConstants.DAPR_SCHEDULER_IMAGE_TAG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import io.dapr.testcontainers.Subscription;
+
+/**
+ * Cross-App Pattern integration test.
+ *
+ * This test demonstrates the cross-app pattern by:
+ * 1. Starting 3 Dapr containers (crossapp-worker, app2, app3)
+ * 2. Launching Java processes that register workflows/activities in separate apps
+ * 3. Executing a cross-app workflow
+ * 4. Asserting successful completion
+ */
+@Testcontainers
+@Tag("testcontainers")
+public class WorkflowsCrossAppCallActivityIT {
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+  
```

Comment:
I am wondering why can't we use `@Container` annotation for all the containers. As far as I can see in the `setUp` method we are just instantiating the containers. Using the `@Container` annotations allows Testcontainer machinery to take care of bootstrap and tear down without explicitly starting or stoping containers.

Using this approach we can get rid of `setUp` and `tearDown` methods.